design/design.md

```diff
@@ -0,0 +1,414 @@
+state of the ledger.
+ALBA can be used to construct a new weighted threshold multisignature scheme
+with a significantly better proof size which
+can allow Cardano nodes to process the chain of Mithril certificates faster.
+If a SNARK is used to compress Mithril certificates to a very small size,
+ALBA can reduce the overall proving time.
+
+Peras and Leios are new extensions of the Ouroboros consensus protocol used in
+Cardano.
+Ouroboros Peras aims to reduce the block settlement time by executing a byzantine fault
+tolerant agreement protocol to decide the next block when adversarial
+conditions are mild.
+The consensus participants certify the chosen block to allow bootstrapping
+nodes to follow the right chain.
+A natural method for this certification is to use a
+weighted threshold multisignature scheme constructed using decentralized ALBA.
+
+Ouroboros Leios aims to increase the throughput of the blockchain in terms of
+the number of transactions per second.
+It does so by decoupling the block payloads from the consensus mechanism which
+operates only on references to blocks.
+To make sure that the chosen block's payload is correct and available
+(which potentially could be not due to malicious behavior),
+all blocks in the input to the consensus protocol need to have
+a certificate of availability proving that at least one honest node has verified
+the block content and is available to distribute it.
+A natural solution to certify data availability would be to use a
+weighted threshold multisignature scheme constructed using decentralized ALBA.
+
+Cross-chain bridges are protocols allowing one blockchain to verify
+information on another blockchain.
+In essense this is the same problem that the Mithril solves with a potentially
+added constraint that on-chain verification must be computationally cheap.
+Whereas Mithril is specific to the Cardano blockchain, one can envision
+a similar protocol, powered by ALBA, for other cryptocurrencies including
+partner chains like Midnight.
+
+Applications of ALBA are not limited to Cardano and blockchains in general.
+For instance, the [paper](https://eprint.iacr.org/2023/1655) describes how
+this primitive can be used to optimize the prover running time of
+a Universally Composable SNARK.
+
+## Requirements
+
+We intend to make an implementation of ALBA that can be used in
+Mithril, Peras, Leios, cross-chain bridges and potentially other protocols
+inside and outside the Cardano universe.
+Our implementation will mostly be guided by our main "clients":
+Mirthril, Peras and Leios.
+
+ALBA is a generic tool that lets one prove knowledge of any kind of data
+elements, but
+as noted in (todo: alba applications section), these three applications
+don't need the full expressivity of ALBA, they really need
+a multisignature scheme, where the data elements would specifically be
+signatures.
+Thus there is question of whether we want our library to provide a general
```

Comment:
No, that is wrong.

We will provide an _example_ that will be Alba used as a multisignature scheme.
This will not be a separate library but a separate crate / folder in this repo.